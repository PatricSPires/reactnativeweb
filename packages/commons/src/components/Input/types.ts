export type InputProps = {
  placeholder?: string
  value?: string
  onChange?: () => void
  disabled?: boolean
  isInvalid: boolean
  label?: string
  helperText?: string
  iconPosition?: 'left' | 'right'
  icon?: React.ReactNode
  iconColor: 'error' | 'warning' | 'success'
  passwordField: boolean
}

export type TextInputProps = Pick<
  InputProps,
  'placeholder' | 'value' | 'onChange' | 'disabled' | 'iconPosition'
>

export type IconProps = Pick<InputProps, 'iconColor' | 'iconPosition'>

export type InputWrapperProps = {
  isInvalid: boolean
}
