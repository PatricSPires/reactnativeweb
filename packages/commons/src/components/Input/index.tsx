import React, { useState } from 'react'
import { View } from 'react-native'
import { EyeOff } from '@styled-icons/feather/EyeOff'
import { Eye } from '@styled-icons/feather/Eye'
import { InputProps } from './types'
import {
  TextInput,
  HelperText,
  Wrapper,
  InputWrapper,
  Icon,
  PasswordIconWrapper,
} from './styles'

const Input = ({
  placeholder,
  value,
  onChange,
  disabled,
  isInvalid,
  helperText,
  iconPosition,
  icon,
  iconColor,
  passwordField,
}: InputProps) => {
  const [showPassword, setShowPassword] = useState(passwordField)

  function changePasswordView() {
    setShowPassword(!showPassword)
  }

  return (
    <Wrapper testID="wrapper">
      {!!helperText && <HelperText>{helperText}</HelperText>}
      <InputWrapper isInvalid={isInvalid}>
        <TextInput
          placeholder={placeholder}
          value={value}
          onChange={onChange}
          disabled={disabled}
          iconPosition={iconPosition}
          secureTextEntry={showPassword}
        />
        {isInvalid && !passwordField && (
          <Icon iconColor={iconColor} iconPosition={iconPosition}>
            {icon}
          </Icon>
        )}
        {passwordField && (
          <PasswordIconWrapper
            testID="passwordIconWrapper"
            onPress={() => changePasswordView()}
          >
            <View>{showPassword ? <EyeOff /> : <Eye />}</View>
          </PasswordIconWrapper>
        )}
      </InputWrapper>
    </Wrapper>
  )
}

Input.defaultProps = {
  passwordField: false,
}

export default Input
