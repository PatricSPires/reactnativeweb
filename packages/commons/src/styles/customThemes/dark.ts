import { DefaultTheme } from 'styled-components'
import { gray, white } from '../colors'
import theme from '../theme'

const dark: DefaultTheme = {
  ...theme,
  title: 'dark',

  colors: {
    ...theme.colors,
    primary_container_background: gray[900],
    medium_contrast_border: gray[700],
    high_contrast_border: gray[500],
    font_color: gray[500],
    primary_color_light: gray[500],
  },
  supportColors: {
    ...theme.supportColors,
    information: white[500],
  },
}

export default dark
