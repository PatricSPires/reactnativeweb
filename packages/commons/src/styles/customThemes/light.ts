import { DefaultTheme } from 'styled-components'
import theme from '../theme'
import { white, gray, blue } from '../colors'

const light: DefaultTheme = {
  ...theme,
  title: 'light',

  colors: {
    primary_container_background: white[100],
    medium_contrast_border: gray[700],
    primary_color_light: blue[700],
    high_contrast_border: gray[500],
    font_color: gray[500],
  },
  supportColors: {
    information: white[500],
  },
}

export default light
