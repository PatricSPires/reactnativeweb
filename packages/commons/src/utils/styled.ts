import styleSheetSerializer from 'jest-styled-components/src/styleSheetSerializer'

export function enableStyledComponentsSerializer() {
  expect.addSnapshotSerializer(styleSheetSerializer)
}
