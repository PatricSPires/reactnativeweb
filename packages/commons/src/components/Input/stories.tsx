import React from 'react'
import { Story } from '@storybook/react/types-6-0'
import { AlertCircle } from '@styled-icons/feather/AlertCircle'
import Input from './index'
import { InputProps } from './types'

export default { title: 'Formul√°rio/Input' }

export const Default: Story<InputProps> = (args) => <Input {...args} />
Default.args = {
  placeholder: 'Placeholder',
  disabled: false,
  isInvalid: false,
}

export const WithLabel: Story<InputProps> = (args) => <Input {...args} />
WithLabel.args = {
  placeholder: 'Placeholder',
  disabled: false,
  isInvalid: false,
  label: 'Enter name',
}

export const WithHelperText: Story<InputProps> = (args) => <Input {...args} />
WithHelperText.args = {
  placeholder: 'Placeholder',
  disabled: false,
  isInvalid: false,
  label: 'Enter name',
  helperText: 'Full name',
}

export const WithErrorIconLeft: Story<InputProps> = (args) => (
  <Input {...args} />
)
WithErrorIconLeft.args = {
  placeholder: 'Placeholder',
  disabled: false,
  isInvalid: true,
  label: 'Enter name',
  helperText: 'Full name',
  icon: <AlertCircle />,
  iconPosition: 'left',
  iconColor: 'error',
}

export const WithErrorIconRight: Story<InputProps> = (args) => (
  <Input {...args} />
)
WithErrorIconRight.args = {
  placeholder: 'Placeholder',
  disabled: false,
  isInvalid: true,
  label: 'Enter name',
  helperText: 'Full name',
  icon: <AlertCircle />,
  iconPosition: 'right',
  iconColor: 'error',
}
