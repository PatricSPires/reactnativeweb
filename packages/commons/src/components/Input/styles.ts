import styled, { css } from 'styled-components'
import {
  Text,
  TextInput as InputRN,
  View,
  TouchableWithoutFeedback,
} from 'react-native'
import { TextInputProps, IconProps, InputWrapperProps } from './types'
import theme from '../../styles/theme'

function getFocusStyles(isInvalid: boolean) {
  if (isInvalid) {
    return `
      outline: 2px solid ${theme.supportColors.error};
      border-bottom-color: transparent;
    `
  }
  return `
    outline: 2px solid ${theme.colors.primary_background};
    border-bottom-color: transparent;
  `
}

export const Wrapper = styled(View)`
  width: 100%;
`

export const InputWrapper = styled(View)<InputWrapperProps>`
  width: 100%;
  display: flex;
  flex-direction: row;
  align-items: center;
  background-color: ${(props) =>
    props.theme.colors.primary_container_background};
  border-bottom: 1px solid ${theme.colors.medium_contrast_border};

  ${({ isInvalid }) => css`
    outline: ${!!isInvalid && `2px solid ${theme.supportColors.error}`};
    border: ${!!isInvalid && 'none'};
  `}

  &:focus-within {
    ${({ isInvalid }) => getFocusStyles(isInvalid)}
  }
`

export const TextInput = styled(InputRN)<TextInputProps>`
  width: 100%;
  height: 40px;
  color: ${(props) => props.theme.colors.font_color};
  font-weight: normal;
  line-height: 18px;
  outline: transparent;
  font-size: 14px;

  order: ${({ iconPosition }) => (iconPosition === 'right' ? 0 : 1)};
  padding: ${({ iconPosition }) =>
    iconPosition === 'left' ? '0 10px 0 0' : '0 0 0 10px'};

  &:disabled {
    outline: none;
    border: none;
    color: ${(props) => props.theme.colors.font_color};
  }

  ::placeholder {
    color: ${(props) => props.theme.colors.font_color};
  }

  &:focus-within {
    color: ${(props) => props.theme.colors.font_color};
  }
`

export const HelperText = styled(Text)`
  font-size: 12px;
  color: ${theme.colors.medium_contrast_border};
`

export const ErrorMessage = styled(Text)`
  font-size: 12px;
  color: ${theme.supportColors.error};
`

export const Icon = styled(View)<IconProps>`
  order: ${({ iconPosition }) => (iconPosition === 'right' ? 1 : 0)};
  color: ${theme.colors.primary_background};
  margin: 0 10px;

  svg {
    width: 16px;
    ${({ iconColor }: IconProps) => css`
      color: ${theme.supportColors[iconColor]};
    `}
  }
`

export const PasswordIconWrapper = styled(TouchableWithoutFeedback)`
  width: 16px;
  height: 16px;
  color: ${(props) => props.theme.colors.font_color};
  margin: 0 10px;
`
