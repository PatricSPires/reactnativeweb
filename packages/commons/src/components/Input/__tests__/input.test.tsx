import * as React from 'react'
import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components'
import { enableStyledComponentsSerializer } from '../../../utils/styled'

import Input from '..'
import dark from '../../../styles/customThemes/dark'
import light from '../../../styles/customThemes/light'

beforeAll(() => {
  enableStyledComponentsSerializer()
})

describe('Input Test', () => {
  it('should render the Input', () => {
    const { toJSON } = render(
      <ThemeProvider theme={dark}>
        <Input isInvalid={false} passwordField={false} />
      </ThemeProvider>,
    )

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render the Input whith label', () => {
    const { toJSON } = render(
      <ThemeProvider theme={dark}>
        <Input label="Hello World" isInvalid={false} passwordField={false} />
      </ThemeProvider>,
    )

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render the Input whith helper text', () => {
    const { toJSON } = render(
      <ThemeProvider theme={dark}>
        <Input
          helperText="Hello World"
          isInvalid={false}
          passwordField={false}
        />
      </ThemeProvider>,
    )

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render the Input with is invalid', () => {
    const { toJSON } = render(
      <ThemeProvider theme={dark}>
        <Input isInvalid passwordField={false} />
      </ThemeProvider>,
    )

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render the Input with passwordField false', () => {
    const { toJSON } = render(
      <ThemeProvider theme={dark}>
        <Input passwordField={false} isInvalid iconColor="error" />
      </ThemeProvider>,
    )

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render the Input with passwordField true', () => {
    const { toJSON } = render(
      <ThemeProvider theme={dark}>
        <Input passwordField isInvalid={false} iconColor="success" />
      </ThemeProvider>,
    )

    expect(toJSON()).toMatchSnapshot()
  })
})

describe('Input Test with dark theme', () => {
  it('should render the Input', () => {
    const { toJSON } = render(
      <ThemeProvider theme={dark}>
        <Input isInvalid={false} iconColor="success" passwordField={false} />
      </ThemeProvider>,
    )

    expect(toJSON()).toMatchSnapshot()
  })
})

describe('Input Test with light theme', () => {
  it('should render the Input', () => {
    const { toJSON } = render(
      <ThemeProvider theme={light}>
        <Input isInvalid={false} iconColor="success" passwordField={false} />
      </ThemeProvider>,
    )

    expect(toJSON()).toMatchSnapshot()
  })
})
