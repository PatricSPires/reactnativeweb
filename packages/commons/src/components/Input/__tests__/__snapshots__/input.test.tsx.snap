// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Test should render the Input 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #161616;
  border-bottom: 1px solid #8d8d8d;
}

.c1:focus-within {
  outline: 2px solid #ffffff;
  border-bottom-color: transparent;
}

.c2 {
  width: 100%;
  height: 40px;
  color: #919191;
  font-weight: normal;
  line-height: 18px;
  outline: transparent;
  font-size: 14px;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  padding: 0 0 0 10px;
}

.c2:disabled {
  outline: none;
  border: none;
  color: #919191;
}

.c2::-webkit-input-placeholder {
  color: #919191;
}

.c2::-moz-placeholder {
  color: #919191;
}

.c2:-ms-input-placeholder {
  color: #919191;
}

.c2::placeholder {
  color: #919191;
}

.c2:focus-within {
  color: #919191;
}

<div
  className="c0 sc-bdnylx c0 css-view-1dbjc4n"
  data-testid="wrapper"
>
  <div
    className="c1 sc-gtssRu c1 css-view-1dbjc4n"
  >
    <input
      autoCapitalize="sentences"
      autoComplete="on"
      autoCorrect="on"
      className="c2 css-textinput-1cwyjr8"
      data-focusable={true}
      dir="auto"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onSelect={[Function]}
      readOnly={false}
      spellCheck={true}
      type="text"
    />
  </div>
</div>
`;

exports[`Input Test should render the Input whith helper text 1`] = `
.c0 {
  width: 100%;
}

.c2 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #161616;
  border-bottom: 1px solid #8d8d8d;
}

.c2:focus-within {
  outline: 2px solid #ffffff;
  border-bottom-color: transparent;
}

.c3 {
  width: 100%;
  height: 40px;
  color: #919191;
  font-weight: normal;
  line-height: 18px;
  outline: transparent;
  font-size: 14px;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  padding: 0 0 0 10px;
}

.c3:disabled {
  outline: none;
  border: none;
  color: #919191;
}

.c3::-webkit-input-placeholder {
  color: #919191;
}

.c3::-moz-placeholder {
  color: #919191;
}

.c3:-ms-input-placeholder {
  color: #919191;
}

.c3::placeholder {
  color: #919191;
}

.c3:focus-within {
  color: #919191;
}

.c1 {
  font-size: 12px;
  color: #8d8d8d;
}

<div
  className="c0 sc-bdnylx c0 css-view-1dbjc4n"
  data-testid="wrapper"
>
  <div
    className="c1 sc-hKFyIo c1 css-text-901oao"
    dir="auto"
  >
    Hello World
  </div>
  <div
    className="c2 sc-gtssRu c2 css-view-1dbjc4n"
  >
    <input
      autoCapitalize="sentences"
      autoComplete="on"
      autoCorrect="on"
      className="c3 css-textinput-1cwyjr8"
      data-focusable={true}
      dir="auto"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onSelect={[Function]}
      readOnly={false}
      spellCheck={true}
      type="text"
    />
  </div>
</div>
`;

exports[`Input Test should render the Input whith label 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #161616;
  border-bottom: 1px solid #8d8d8d;
}

.c1:focus-within {
  outline: 2px solid #ffffff;
  border-bottom-color: transparent;
}

.c2 {
  width: 100%;
  height: 40px;
  color: #919191;
  font-weight: normal;
  line-height: 18px;
  outline: transparent;
  font-size: 14px;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  padding: 0 0 0 10px;
}

.c2:disabled {
  outline: none;
  border: none;
  color: #919191;
}

.c2::-webkit-input-placeholder {
  color: #919191;
}

.c2::-moz-placeholder {
  color: #919191;
}

.c2:-ms-input-placeholder {
  color: #919191;
}

.c2::placeholder {
  color: #919191;
}

.c2:focus-within {
  color: #919191;
}

<div
  className="c0 sc-bdnylx c0 css-view-1dbjc4n"
  data-testid="wrapper"
>
  <div
    className="c1 sc-gtssRu c1 css-view-1dbjc4n"
  >
    <input
      autoCapitalize="sentences"
      autoComplete="on"
      autoCorrect="on"
      className="c2 css-textinput-1cwyjr8"
      data-focusable={true}
      dir="auto"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onSelect={[Function]}
      readOnly={false}
      spellCheck={true}
      type="text"
    />
  </div>
</div>
`;

exports[`Input Test should render the Input with is invalid 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #161616;
  border-bottom: 1px solid #8d8d8d;
  outline: 2px solid #DA1E28;
  border: none;
}

.c1:focus-within {
  outline: 2px solid #DA1E28;
  border-bottom-color: transparent;
}

.c2 {
  width: 100%;
  height: 40px;
  color: #919191;
  font-weight: normal;
  line-height: 18px;
  outline: transparent;
  font-size: 14px;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  padding: 0 0 0 10px;
}

.c2:disabled {
  outline: none;
  border: none;
  color: #919191;
}

.c2::-webkit-input-placeholder {
  color: #919191;
}

.c2::-moz-placeholder {
  color: #919191;
}

.c2:-ms-input-placeholder {
  color: #919191;
}

.c2::placeholder {
  color: #919191;
}

.c2:focus-within {
  color: #919191;
}

.c3 {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  color: #ffffff;
  margin: 0 10px;
}

.c3 svg {
  width: 16px;
  color: #DA1E28;
}

<div
  className="c0 sc-bdnylx c0 css-view-1dbjc4n"
  data-testid="wrapper"
>
  <div
    className="c1 sc-gtssRu c1 css-view-1dbjc4n"
  >
    <input
      autoCapitalize="sentences"
      autoComplete="on"
      autoCorrect="on"
      className="c2 css-textinput-1cwyjr8"
      data-focusable={true}
      dir="auto"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onSelect={[Function]}
      readOnly={false}
      spellCheck={true}
      type="text"
    />
    <div
      className="c3 sc-jSFkmK c3 css-view-1dbjc4n"
    />
  </div>
</div>
`;

exports[`Input Test should render the Input with passwordField false 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #161616;
  border-bottom: 1px solid #8d8d8d;
  outline: 2px solid #DA1E28;
  border: none;
}

.c1:focus-within {
  outline: 2px solid #DA1E28;
  border-bottom-color: transparent;
}

.c2 {
  width: 100%;
  height: 40px;
  color: #919191;
  font-weight: normal;
  line-height: 18px;
  outline: transparent;
  font-size: 14px;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  padding: 0 0 0 10px;
}

.c2:disabled {
  outline: none;
  border: none;
  color: #919191;
}

.c2::-webkit-input-placeholder {
  color: #919191;
}

.c2::-moz-placeholder {
  color: #919191;
}

.c2:-ms-input-placeholder {
  color: #919191;
}

.c2::placeholder {
  color: #919191;
}

.c2:focus-within {
  color: #919191;
}

.c3 {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  color: #ffffff;
  margin: 0 10px;
}

.c3 svg {
  width: 16px;
  color: #DA1E28;
}

<div
  className="c0 sc-bdnylx c0 css-view-1dbjc4n"
  data-testid="wrapper"
>
  <div
    className="c1 sc-gtssRu c1 css-view-1dbjc4n"
  >
    <input
      autoCapitalize="sentences"
      autoComplete="on"
      autoCorrect="on"
      className="c2 css-textinput-1cwyjr8"
      data-focusable={true}
      dir="auto"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onSelect={[Function]}
      readOnly={false}
      spellCheck={true}
      type="text"
    />
    <div
      className="c3 sc-jSFkmK c3 css-view-1dbjc4n"
    />
  </div>
</div>
`;

exports[`Input Test should render the Input with passwordField true 1`] = `
.c4 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  width: 100%;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #161616;
  border-bottom: 1px solid #8d8d8d;
}

.c1:focus-within {
  outline: 2px solid #ffffff;
  border-bottom-color: transparent;
}

.c2 {
  width: 100%;
  height: 40px;
  color: #919191;
  font-weight: normal;
  line-height: 18px;
  outline: transparent;
  font-size: 14px;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  padding: 0 0 0 10px;
}

.c2:disabled {
  outline: none;
  border: none;
  color: #919191;
}

.c2::-webkit-input-placeholder {
  color: #919191;
}

.c2::-moz-placeholder {
  color: #919191;
}

.c2:-ms-input-placeholder {
  color: #919191;
}

.c2::placeholder {
  color: #919191;
}

.c2:focus-within {
  color: #919191;
}

.c3 {
  width: 16px;
  height: 16px;
  color: #919191;
  margin: 0 10px;
}

<div
  className="c0 sc-bdnylx c0 css-view-1dbjc4n"
  data-testid="wrapper"
>
  <div
    className="c1 sc-gtssRu c1 css-view-1dbjc4n"
  >
    <input
      autoCapitalize="sentences"
      autoComplete="on"
      autoCorrect="on"
      className="c2 css-textinput-1cwyjr8"
      data-focusable={true}
      dir="auto"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onSelect={[Function]}
      readOnly={false}
      spellCheck={true}
      type="password"
    />
    <div
      className="c3 sc-gKAblj c3 css-view-1dbjc4n r-cursor-1loqt21 r-touchAction-1otgn73"
      data-focusable={true}
      data-testid="passwordIconWrapper"
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      tabIndex="0"
    >
      <svg
        aria-hidden="true"
        className="c4"
        fill="none"
        focusable="false"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17.94 17.94A10.07 10.07 0 0112 20c-7 0-11-8-11-8a18.45 18.45 0 015.06-5.94M9.9 4.24A9.12 9.12 0 0112 4c7 0 11 8 11 8a18.5 18.5 0 01-2.16 3.19m-6.72-1.07a3 3 0 11-4.24-4.24"
        />
        <line
          x1={1}
          x2={23}
          y1={1}
          y2={23}
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Input Test with dark theme should render the Input 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #161616;
  border-bottom: 1px solid #8d8d8d;
}

.c1:focus-within {
  outline: 2px solid #ffffff;
  border-bottom-color: transparent;
}

.c2 {
  width: 100%;
  height: 40px;
  color: #919191;
  font-weight: normal;
  line-height: 18px;
  outline: transparent;
  font-size: 14px;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  padding: 0 0 0 10px;
}

.c2:disabled {
  outline: none;
  border: none;
  color: #919191;
}

.c2::-webkit-input-placeholder {
  color: #919191;
}

.c2::-moz-placeholder {
  color: #919191;
}

.c2:-ms-input-placeholder {
  color: #919191;
}

.c2::placeholder {
  color: #919191;
}

.c2:focus-within {
  color: #919191;
}

<div
  className="c0 sc-bdnylx c0 css-view-1dbjc4n"
  data-testid="wrapper"
>
  <div
    className="c1 sc-gtssRu c1 css-view-1dbjc4n"
  >
    <input
      autoCapitalize="sentences"
      autoComplete="on"
      autoCorrect="on"
      className="c2 css-textinput-1cwyjr8"
      data-focusable={true}
      dir="auto"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onSelect={[Function]}
      readOnly={false}
      spellCheck={true}
      type="text"
    />
  </div>
</div>
`;

exports[`Input Test with light theme should render the Input 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #ffffff;
  border-bottom: 1px solid #8d8d8d;
}

.c1:focus-within {
  outline: 2px solid #ffffff;
  border-bottom-color: transparent;
}

.c2 {
  width: 100%;
  height: 40px;
  color: #919191;
  font-weight: normal;
  line-height: 18px;
  outline: transparent;
  font-size: 14px;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  padding: 0 0 0 10px;
}

.c2:disabled {
  outline: none;
  border: none;
  color: #919191;
}

.c2::-webkit-input-placeholder {
  color: #919191;
}

.c2::-moz-placeholder {
  color: #919191;
}

.c2:-ms-input-placeholder {
  color: #919191;
}

.c2::placeholder {
  color: #919191;
}

.c2:focus-within {
  color: #919191;
}

<div
  className="c0 sc-bdnylx c0 css-view-1dbjc4n"
  data-testid="wrapper"
>
  <div
    className="c1 sc-gtssRu c1 css-view-1dbjc4n"
  >
    <input
      autoCapitalize="sentences"
      autoComplete="on"
      autoCorrect="on"
      className="c2 css-textinput-1cwyjr8"
      data-focusable={true}
      dir="auto"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onSelect={[Function]}
      readOnly={false}
      spellCheck={true}
      type="text"
    />
  </div>
</div>
`;
